bin/metadata_validation.py

```diff
@@ -1,55 +1,61 @@
 #!/usr/bin/env python3
 
-import os
 import sys
+import re
 
 import json
 
 import synapseclient
-from synapseclient import Entity
 
 
-def get_synaspe_file(syn_id):
+def get_synaspe_file(syn_id: str) -> synapseclient.entity.File:
     """
-    Initializes synapseclient session and retrieves file metadata
+    1. Validates provided Synapse ID against regex pattern
+    2. Initializes synapseclient session and retrieves file metadata
     """
-    syn = synapseclient.Synapse()
-    syn.login(authToken=os.environ["SYNAPSE_AUTH_TOKEN"])
-    syn_file = syn.get(syn_id)
+    pattern = re.compile("syn[0-9]+")
+    assert pattern.search(syn_id), f"{syn_id} is an invalid Synapse ID"
```

Comment:
I came here to suggest that you can add a message to your assertion error, but you already did that. 

I'll note that on my PRs, @thomasyu888 has been suggesting that `assert` statements be restricted to tests. I'm indifferent about it, so I've been changing my `assert` statements to something like this: 

```suggestion
    if not pattern.search(syn_id):
        raise ValueError(f"{syn_id} is an invalid Synapse ID")
```